### ISSUE TITLE
procstat input plugin has typo in example configuration; supervisor_units

### ISSUE THREAD

**Issue by jackson-ball** [influxdata/telegraf#14327]
### Relevant telegraf.conf

```toml
[[inputs.procstat]]
supervisor_units = ["program_name"]
```

### Logs from Telegraf

```text
2023-11-21T03:28:16Z I! Loading config: procstat.conf
2023-11-21T03:28:16Z E! error loading config file procstat.conf: plugin inputs.procstat: line 2: configuration specified the fields ["supervisor_units"], but they weren't used
```

### System info

Telegraf 1.29.0-4c4b821a (git: master@4c4b821a), Redhat, 4.18.0-477.27.1.el8_8.x86_64

### Steps to reproduce

1. Using this sample config
```
[[inputs.procstat]]
supervisor_units = ["program_name"]
```
2. Running telegraf with `telegraf --test --config procstat.conf` 
3. Fails with:
```
2023-11-21T03:28:16Z E! error loading config file procstat.conf: plugin inputs.procstat: line 2: configuration specified the fields ["supervisor_units"], but they weren't used
```
4. Using this sample config:
```
[[inputs.procstat]]
supervisor_unit = ["program_name"]
```
5. Running telegraf with `telegraf --test --config procstat.conf` runs as expected with no errors
...

### Expected behavior

Expect that `supervisor_units` is the correct syntax for the procstat plugin as listed in the example config file.

### Actual behavior

It seems as though `supervisor_unit` is the correct usage.

**Comment by srebhan** [comment]
@jackson-ball can you please test the binary in #14329 available once CI finished all tests!? Let me know if this fixes the issue!

**Comment by jackson-ball** [comment]
Hi @srebhan thanks for taking a look at this issue. Yes I tested the Linux amd64 binary and confirmed this fixes the issue.

**Comment by srebhan** [comment]
@jackson-ball please keep the issue open, it will be automatically closed once the PR is merged into master...
